### DISCUSSION TITLE
Undo close pane: remember pin status

### DISCUSSION THREAD

**Discussion by FelipeRearden** [Taitava/obsidian-shellcommands#87] [Offtopic]
Hello @Taitava !

I just saw your post of the forum about this topic. 

100% agree with you that we need a native solution for this. 

While we wait, I think i found two workarounds to automate this as a only command: 

## Workaround 01 - QuickAdd Macro with the following commands:

1- Undo Close Panes
2- Cycle Through Panes: Focus on The Last Active Pane [[require Cycle Through Panes plug-in](https://github.com/phibr0/cycle-through-panes)]
3- Toggle Pin  

## Workaround 02 - SC + Obsidian URI

Create a SC with these three commands using Obsidian URI

```
open "obsidian://advanced-uri?vault=BASB&commandid=XXXXXXX" && open "obsidian://advanced-uri?vault=BASB&commandid=workspace%253Atoggle-pin" &&open "obsidian://advanced-uri?vault=BASB&commandid=workspace%253Atoggle-pin"
```

I'm on 0.12.15 yet and I don't know the command id for the Undo Close Pane

---

Have a great day :)

**Comment by Taitava** [comment]
Thanks! 🙂

**Comment by Taitava** [comment]
> I'm on 0.12.15 yet and I don't know the command id for the Undo Close Pane

If you haven't yet upgraded, I can recommend to upgrade - [Obsidian `0.12.17`](https://forum.obsidian.md/t/obsidian-release-v0-12-17/25270) added a feature that if you type `/` in editor (need to have a space or something before `/`), Obsidian will open command palette for you, so it's even quicker than hitting Ctrl/Cmd + P. It's a new core plugin, so you have to enable it manually (_Slash commands_).

**Reply by FelipeRearden** [reply]
> > I'm on 0.12.15 yet and I don't know the command id for the Undo Close Pane
> 
> 
> 
> If you haven't yet upgraded, I can recommend to upgrade - [Obsidian `0.12.17`](https://forum.obsidian.md/t/obsidian-release-v0-12-17/25270) added a feature that if you type `/` in editor (need to have a space or something before `/`), Obsidian will open command palette for you, so it's even quicker than hitting Ctrl/Cmd + P. It's a new core plugin, so you have to enable it manually (_Slash commands_).

👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 👀 

This is something very important to keyboard lovers like me !!!!

Thanks @Taitava !!! I wanna take a look at :)

**Reply by Taitava** [reply]
Nice! Btw speaking of loving keyboards, may I ask what kind of keyboard are you using, something manufactured by Apple or something else? I'm thinking about upgrading mine maybe next year. I'm thinking about a split keyboard, but then again I might consider other options too if they are ergonomic.

**Reply by FelipeRearden** [reply]
> Nice! Btw speaking of loving keyboards, may I ask what kind of keyboard are you using, something manufactured by Apple or something else? I'm thinking about upgrading mine maybe next year. I'm thinking about a split keyboard, but then again I might consider other options too if they are ergonomic.

I use the Logitech K780.
- https://www.logitech.com/en-us/products/keyboards/k780-multi-device-wireless-keyboard.920-008149.html

And the Logitech M720 Thriatlon as my mouse;
- I'm crazy for an Magic Trackpad but is extremely expensive; 

BUT, I have an special reason: since I use my iPad as a laptop, I need a keyboard that have:
- Multiple connections (3 bluetooth channels)
- a built-in stand to place the iPad 12.9
- iOS dedicated keys 


What I don't like: the keys are circle/ellipsis instead of squared. It took a lot of time to get used to because I like the standard keys. 

It's a fantastic keyboard (almost 5 years old) if you are willing to get used to the keys :)

**Reply by Taitava** [reply]
I have a seven years old Ducky Shine4 mechanical keyboard for doing work on desktop. For mobile devices, I have a [Logitech K480 keyboard](https://www.logitech.com/fi-fi/products/keyboards/k480-multi-device-wireless.html) whose keys are a bit rounded, but not circle. It has Mac keys, too, which is good as one of my mobile devices is an iPhone.

**Reply by FelipeRearden** [reply]
Nice! I like Logitech keyboards. 

I would like to try the slip keyboard as well!

**Reply by Taitava** [reply]
Yea, Logitech makes good quality keyboards and mice. I've found their wireless technology to be quite stable, works well.
